src/main/java/com/routeme/service/SearchServiceImpl.java

```diff
@@ -80,51 +116,186 @@ private SearchResponseDTO convertRouteDTOsToSearchResponseDTO(List<RouteDTO> rou
         routeDTO.setTransportationModes(route.getTransportationModes());
-        routeDTO.setSteps(route.getStepSummaries());
+        setRouteDTOStepDTOs(routeDTO, route);
         if (route instanceof TransitRoute) {
-            DateFormat timeFormat = new SimpleDateFormat("KK:mm a");
-            routeDTO.setArrivalTime(timeFormat.format(route.getArrivalTime().toDate()));
-            routeDTO.setDepartureTime(timeFormat.format(route.getDepartureTime().toDate()));
+            routeDTO.setArrivalTime(Util.Route.TIME_FORMAT.format(route.getArrivalTime().toDate()));
+            routeDTO.setDepartureTime(Util.Route.TIME_FORMAT.format(route.getDepartureTime().toDate()));
+            routeDTO.setDepartureDateTimeInMillis(route.getDepartureTime().getMillis());
             routeDTO.setRouteSummary(routeDTO.getDepartureTime() + "-" + routeDTO.getArrivalTime() + " ("
                     + routeDTO.getDuration() + ")");
             routeDTO.setTransit(true);
+            routeDTO.setDurationInSeconds(route.getDuration().inSeconds);
+            routeDTO.setNumberOfChanges(((TransitRoute) route).getNumberOfChanges());
         } else {
             routeDTO.setRouteSummary(route.getRouteSummary() + " (" + routeDTO.getDistance() + ")");
             routeDTO.setTransit(false);
         }
         return routeDTO;
     }
 
+    private void setRouteDTOStepDTOs(RouteDTO routeDTO, Route route) {
+        ArrayList<StepDTO> stepDTOs = new ArrayList<StepDTO>();
+        for (Step step : route.getStepsData()) {
+            StepDTO stepDTO = convertStepToStepDTO(step);
+            stepDTOs.add(stepDTO);
+        }
+        routeDTO.setSteps(stepDTOs);
+    }
+
+    private StepDTO convertStepToStepDTO(Step step) {
+        StepDTO stepDTO = null;
+        if (step instanceof TransitStep) {
+            TransitStep transitStep = (TransitStep) step;
+            String statTime = Util.Route.TIME_FORMAT.format(transitStep.getStartTime().toDate());
+            String endTime = Util.Route.TIME_FORMAT.format(transitStep.getEndTime().toDate());
+            stepDTO = new TransitStepDTO(transitStep.getTransportationMode(), transitStep.getStartStation(),
+                    transitStep.getEndStation(), transitStep.getDuration(), statTime, endTime,
+                    transitStep.getTransportationVehicleShortName(), transitStep.getTransportationLineColorCode(),
+                    transitStep.getTransportationLineHeadSign());
+        } else {
+            NonTransitStep nonTransitStep = (NonTransitStep) step;
+            stepDTO = new NonTransitStepDTO(nonTransitStep.getTransportationMode(), nonTransitStep.getHtmlIntruction(),
+                    nonTransitStep.getDistance());
+        }
+        return stepDTO;
+    }
+}
+
+class RoutePreferenceSorter implements Comparator<RouteDTO> {
+    ArrayList<String> travelModePreference = new ArrayList<String>();
+    ArrayList<String> routeTypePreference = new ArrayList<String>();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/heshamhamdymassoud/RouteMe-API/file/2647790198/issues/source?bid=3287636&fileBranchId=3323263#l227)